how-tos/maple.md

```diff
@@ -0,0 +1,182 @@
+# maple
+
+Last updated 2017-10-11
+
+Currently, maple is our taskcluster relpro migration area.
+
+Aki's on PTO 2017-10-12 - 2017-10-13, so here's a brain dump.
+
+---
+## Current status
+
+### Trello
+
+The [Trello board](https://trello.com/b/EGWsGSXT/tc-migration-release-h2-2017) has the latest human tasks. `Fixed on maple` currently means we're able to generate the right graph... we'll have additional work to get this green end-to-end.
+
+Please feel free to add tasks there if we're missing any!
+
+### open bugs
+
+- [bug 1407817](https://bugzilla.mozilla.org/show_bug.cgi?id=1407817) has some missing buildbot-configs and in-tree patches for fennec relpro on maple.
+
+Random thoughts:
+- Should we switch our properties to dashed-words instead of `underscore_words`?
+- Should we rename `candidates_fennec` to `promote_fennec` ?
+
+---
+
+### Dep vs release
+
+Right now, we're set up differently than jamun, in that we're signing with dep scopes. There are a couple approaches we can take:
+
+1. Proceed with dep scopes, and fix anything that prevents us from doing an end-to-end staging release
+1. Turn on release signing, and fix staging releases with dep scopes later.
+
+#### Dep scopes
+
+If we proceed with dep scopes,
+
+- we will need dep scriptworker pools for {beetmover,balrog,pushapk}. These don't have to be large. They'll have dep creds only, and will verify chain of trust artifacts, but not the cot signatures. This is because dep signing doesn't sign its cot artifacts.
+- we'll have to deal with update verify and other mar signature checks
+- we'll have to patch pushapk to enable dep scope -- this will be a dry-run, and we could check for the dep apk signature.
+
+This have more work items, but we'll end up with a staging release that we can run anywhere, even try!
+
+#### Release scopes
+
+- We'll have to deal with maple not having the release signing scopes -- just a taskcluster bug away
+- we'll have to update taskcluster/taskgraph/util/scriptworker.py to enable release signing
+
+This may be less work, but we'll be further away from our end goal when we're done.
+
+---
+
+### Scriptworker or encrypted env vars
+
+We can either use beetmover scriptworker + balrog scriptworker, or keep using encrypted env vars.
+
+#### beetmover+balrog scriptworker
+
+- balrog scriptworker needs release support
+- beetmover scriptworker needs push to releases support
+- we have to cot-enable *everything* upstream of these tasks. We have balrog+beetmover tasks in the recompression tasks way at the end of the graph -- they'll need to be part of the chain!
+
+This may be more work, but avoids some hacks and is where we want to end up.
+
+#### encrypted env vars
+
+- we can probably add some inputs into the relpro action for env vars, allowing us to generate these before submitting the action task.
```

Comment:
We would need to know the corresponding task IDs - you have to use them to make encrypted env vars work.

Reply:
Ugh. Then we shouldn't use the encrypted env vars =\